codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientClass.java

```diff
@@ -442,9 +445,27 @@ protected static MethodSpec updateSdkClientConfigurationMethod(
                .addStatement("$1T serviceConfigBuilder = new $1T(configuration)", serviceClientConfigurationBuilderClassName)
                .beginControlFlow("for ($T plugin : plugins)", SdkPlugin.class)
                .addStatement("plugin.configureClient(serviceConfigBuilder)")
-               .endControlFlow()
-               .addStatement("return configuration.build()");
-
+               .endControlFlow();
```

Comment:
I only modified the Sync class and did not modify the Async class as it would suffice to incorporate the "request-level" modification check into the DefaultServiceClient Client